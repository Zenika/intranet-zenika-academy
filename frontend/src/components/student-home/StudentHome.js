import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import './StudentHome.scss';

class StudentHome extends Component {
  constructor(props) {
    super(props);
    this.state = {
      redirectToUser: false,
      promotionId: '',
    };
  }

  componentDidMount() {
    document.title = "Page d'accueil Eleve";
    const id = sessionStorage.getItem('promoId');
    const role = sessionStorage.getItem('userRole');
    if (JSON.parse(role) === 2 || JSON.parse(role) === 3) {
      return this.setState({
        redirectToUser: true,
        promotionId: id,
      });
    }
    return true;
  }

  render() {
    const { redirectToUser, promotionId } = this.state;
    if (redirectToUser) {
      return <Redirect to={`/user/promo/${promotionId}/details`} />;
    }
    return <h1>Hello!</h1>;
  }
}

export default StudentHome;
