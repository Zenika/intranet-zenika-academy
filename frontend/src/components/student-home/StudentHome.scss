.studentHomeSection{
  margin: 1rem 0
}

.studentsHomeSectionTitle{
  font-weight: bold;
}