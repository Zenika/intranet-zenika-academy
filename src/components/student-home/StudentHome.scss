.detailsLink {
  color:blue
}

.promoLineContainer{
  margin:.5rem
}

.promosContainer{
  display: flex;
  flex-direction: column;
}
