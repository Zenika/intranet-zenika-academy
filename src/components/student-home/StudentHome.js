import React, { Component } from 'react';
import axios from 'axios';
import Moment from 'react-moment';
import './StudentHome.scss';

class StudentHome extends Component {
  _isMounted = false;

  constructor(props) {
    super(props);
    this.state = { promotionDetails: [] };
  }

  componentDidMount() {
    const id = sessionStorage.getItem('promoId');
    this._isMounted = true;
    axios.get(`http://localhost:4000/api/promotions/details/${id}`)
      .then((res) => {
        const promotionDetails = res.data;
        if (this._isMounted) {
          this.setState({ promotionDetails });
        }
      });
  }

  componentWillUnmount() {
    this._isMounted = false;
  }

  displayProgram = () => {
    const program = [];
    const { promotionDetails } = this.state;
    const programData = promotionDetails.program ? promotionDetails.program : [];
    const keys = Object.keys(programData);
    program.push(<h1 className="studentsHomeSectionTitle" key="Program Title">Programme : </h1>);
    keys.forEach((key) => {
      if (key === 'title') {
        program.push(
          <div key={key}>
Titre :
            { ' ' }
            {programData[key]}
          </div>,
        );
      }
    });
    return program;
  }

  displayPromotion = () => {
    const promotion = [];
    const { promotionDetails } = this.state;
    const promotionData = promotionDetails.promotion ? promotionDetails.promotion : [];
    const keys = Object.keys(promotionData);
    promotion.push(<h1 className="studentsHomeSectionTitle" key="promoTitle">Promotion : </h1>);
    keys.forEach((key) => {
      if (key === 'title') {
        promotion.push(
          <div key={key}>
Titre :
            { ' ' }
            {promotionData[key]}
          </div>,
        );
      }
      if (key === 'startDate') {
        promotion.push(
          <div key={key}>
DÃ©but :
            { ' ' }
            <Moment format="DD/MM/YYYY" key={key}>{promotionData[key]}</Moment>
          </div>,
        );
      }
      if (key === 'endDate') {
        promotion.push(
          <div key={key}>
Fin :
            { ' ' }
            <Moment format="DD/MM/YYYY" key={key}>{promotionData[key]}</Moment>
          </div>,
        );
      }
      if (key === 'city') {
        promotion.push(
          <div key={key}>
Ville :
            { ' ' }
            {promotionData[key]}
          </div>,
        );
      }
    });
    return promotion;
  }

  displayStudents = () => {
    const students = [];
    const { promotionDetails } = this.state;
    const usersData = promotionDetails.users ? promotionDetails.users : [];
    students.push(<h1 className="studentsHomeSectionTitle" key="promoStudentsTitle">Eleves : </h1>);
    usersData.forEach((user) => {
      if (user.role === 3) {
        students.push(
          <div key={user.id}>
            {user.firstName}
            { ' ' }
            {user.lastName.toUpperCase()}
          </div>,
        );
      }
    });
    return students;
  }

  displayTeachers = () => {
    const teacher = [];
    const { promotionDetails } = this.state;
    const usersData = promotionDetails.users ? promotionDetails.users : [];
    teacher.push(<h1 className="studentsHomeSectionTitle" key="promoteacherTitle">Formateurs : </h1>);
    usersData.forEach((user) => {
      if (user.role === 2) {
        teacher.push(
          <div key={user.id}>
            {user.firstName}
            { ' ' }
            {user.lastName.toUpperCase()}
          </div>,
        );
      }
    });
    return teacher;
  }

  render() {
    return (
      <>
        <h1 className="title is-2 mbmd">Bienvenue</h1>
        <section className="studentHomeSection">{this.displayPromotion()}</section>
        <section className="studentHomeSection">{this.displayStudents()}</section>
        <section className="studentHomeSection">{this.displayProgram()}</section>
        <section className="studentHomeSection">{this.displayTeachers()}</section>
      </>
    );
  }
}

export default StudentHome;
