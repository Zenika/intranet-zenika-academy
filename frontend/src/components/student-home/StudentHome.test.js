import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import StudentHome from './StudentHome';

Enzyme.configure({ adapter: new Adapter() });

describe('StudentHome component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<StudentHome />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('matches snapshot', () => {
    const component = renderer.create(<StudentHome />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
