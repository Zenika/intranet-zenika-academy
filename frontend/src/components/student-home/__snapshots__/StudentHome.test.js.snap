// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentHome component matches snapshot 1`] = `
<h1>
  Hello!
</h1>
`;
